src/box/lua/config/configdata.lua

```diff
@@ -119,6 +214,19 @@ local function new(iconfig, cconfig, instance_name)
     local found = cluster_config:find_instance(cconfig, instance_name)
     assert(found ~= nil)
 
+    local replicaset_uuid = instance_config:get(iconfig_def,
+        'database.replicaset_uuid')
+    local instance_uuid = instance_config:get(iconfig_def,
+        'database.instance_uuid')
+    if instance_config:get(iconfig_def, 'sharding') ~= nil then
+        if replicaset_uuid == nil then
+            replicaset_uuid = uuid_from_name(found.replicaset_name)
+        end
+        if instance_uuid == nil then
+            instance_uuid = uuid_from_name(instance_name)
+        end
+    end
```

Comment:
I think that we shouldn't assign UUIDs depending on a presence of the `sharding` options. One may want to start a bunch of instances and later assign them sharding roles.

Reply:
Done.